src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/Utils.scala

```diff
@@ -307,4 +308,30 @@ object Utils {
         }
     } yield ()
   }
+
+  def pollEServiceById(
+    fetchFunc: => Future[CatalogProcess.EService],
+    condition: CatalogProcess.EService => Boolean,
+    maxRetries: Int = 10,
+    delay: FiniteDuration = 1.second
+  )(implicit ec: ExecutionContext): Future[Unit] = {
+
+    def poll(attempt: Int): Future[Unit] = {
+      if (attempt > maxRetries) {
+        Future.failed(new RuntimeException("Max retries reached"))
+      } else {
+        fetchFunc.flatMap { result =>
```

Comment:
`fetchFunc` may fail when `catalogProcessService.getEServiceById` is passed (`getEServiceById` returns 404 if the eservice is not found yet).
The retry should consider also the error case using `recoverWith`